packages/node-core/src/indexer/dictionary/types.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {IBlock} from '@subql/types-core';
+import {BlockHeightMap} from '../../utils/blockHeightMap';
+
+export type DictionaryResponse<B = number> = {
+  batchBlocks: B[];
+  lastBufferedHeight: number;
+};
+
+export enum DictionaryVersion {
+  v1 = 'v1',
+  v2Basic = 'v2Basic',
+  v2Complete = 'v2Complete',
+}
```

Comment:
This needs to be removed. Beyond the base implementations of v1/v2 dictionaries the dictionary service should not have any concept of types or versions of dictionaries. 

This limits implementing any sort of other custom dictionary that is specific to an SDK.